template/template.go

```diff
@@ -451,7 +451,9 @@ func (e *executer) resolvePreferences(m ast.Matcher, res []any) [][]string {
 				switch key := vKey.(type) {
 				case ast.CatchAllKey:
 					continue
-				case ast.NameLiteral, ast.QuotedLiteral, ast.NumberLiteral:
+				case ast.QuotedLiteral:
+					keys = append(keys, string(key))
+				case ast.NameLiteral, ast.NumberLiteral:
 					keys = append(keys, key.String())
```

Comment:
Based on our conversation, this should be something like, right?

```go
case ast.QuotedLiteral, ast.NameLiteral:
  keys = append(keys, string(key))
case ast.NumberLiteral:
  keys = append(keys, key.String())
```

Reply:
Proposed variant complains about `cannot convert key (variable of type parse.VariantKey) to type string`. I'll leave it as is and will add a comment.

Reply:
Moved `QuotedLiteral` and `NameLiteral` to separated cases to improve code reading.